katdal/dataset.py

```diff
@@ -658,7 +666,9 @@ def select(self, **kwargs):
             value via a sequence of string pairs, or select all autocorrelations
             via 'auto' or all cross-correlations via 'cross'.
         ants : string or :class:`katpoint.Antenna` object or sequence, optional
-            Select antennas by name or object
+            Select antennas by name or object. If all antennas specified are
```

Comment:
I haven't considered the issue of mixing '~' with normal selections, which is kind of pointless :-)

Maybe we should consider just having a single tilde at the start to negate the entire selection?

Reply:
Yeah, I think that is the only case that makes sense. It would be pointless to deselect some antenna and then select others... 
In terms of interface consistency, will we be able to treat _all_ the deselects in the same way i.e. `select(kwarg=selections, kwarg2=~deselections...)`

Reply:
Good question... The only deselection currently occurs in `scans` and `compscans`. I see that it also uses `_selection_to_list` and checks individual terms for a leading tilde. It therefore expects e.g. `'~slew,~stop'`.

But then it dutifully goes and ORs those two selections together... 🤦🏾 So it only really makes sense for one negative term at a time. We could also fix that to be a single leading tilde too.

Bear in mind that katdal 1.0 will happen sometime this year, which will provide the opportunity to fix any warts.